migration/migration.go

```diff
@@ -16,23 +17,38 @@ import (
 	"github.com/NethermindEth/juno/encoder"
 	"github.com/NethermindEth/juno/utils"
 	"github.com/bits-and-blooms/bitset"
+	"github.com/fxamacker/cbor/v2"
 	"github.com/sourcegraph/conc/pool"
 )
 
+type schemaMetadata struct {
+	Version           uint64
+	IntermediateState []byte
+}
+
+func (schema *schemaMetadata) fromBytes(data []byte) error {
```

Comment:
I think the idiomatic name for these methods would be `MarshalBinary` and `UnmarshalBinary`. Let's use those if you agree.